ftp/1.5/modules/ROOT/pages/ftp-dynamic-connection.adoc

```diff
@@ -0,0 +1,155 @@
+** On each *For Each* iteration, the expressions set in the FTP global configuration element resolve to a different value and establish different connections to each of the servers.
+
+To test this example, create the Mule application, and run and test the application with curl commands:
+
+image::ftp_dynamic_connect_flow.png[FTP Servers Dynamic Connection Flow in Studio Canvas]
+
+== Create the Mule Application
+
+To create the Mule flow:
+
+. In the *Mule Palette* view, select *HTTP > Listener*.
+. Drag *HTTP Listener* to the Studio canvas.
+. On the *HTTP Listener* configuration screen, optionally change the value of the *Display Name* field.
+. Set *Path* to `/multitenant`.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of HTTP Listener in the app.
+. On the *General* tab, set the following fields:
++
+* *Host*: `All Interfaces [0.0.0.0] (default)`
+* *Port*: `8081`
++
+. Click *OK*.
+. Drag a *Set Variable* component to the right of the *HTTP Listener* source.
+. Set *Name* to `content` and *Value* to `#[payload]`.
+. Drag the File Connector *Read* operation to the right of the *Set Variable* component.
+. Set the *File Path* field to `recipients.csv`. +
+  This reads a CSV file that contains a random set of FTP destinations with columns such as `host`, `user`, and `port`.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of File Connector in the app.
+. Accept the default settings and click *OK*.
+. On the *MIME Type* tab of the *Read* operation, for *MIME Type*, select *application/csv*.
+
+. In Studio, drag a *Transform Message* component to the right of the *Read* operation.
+. In the *Output* section of the component, add the following DataWeave code to map the columns of the CSV file:
++
+.DataWeave script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/java
+---
+payload map using (row = $) {
+   host: row.Host,
+   user: row.User,
+   password: row.Password}
+----
++
+. Drag a *For Each* component to the right of the *Transform Message* component.
+. Drag an FTP *Write* operation into the *For Each* component. +
+On each For Each iteration, the *Write* operation writes the contents into each of the FTP destinations.
+. Click the plus sign (*+*) next to the *Connector configuration* field to configure a global element that can be used by all instances of FTP *Write* operation in the app, and set the following fields:
++
+* *Host*: `#[payload.host]`
+* *Username*: `#[payload.username]`
+* *Password*: `#[payload.password]`
++
+[start=6]
+. Click *OK*.
+. Set *Path* to `demo.txt`.
+. Set the *Content* field to `payload`.
+. Drag a *Set Payload* component to the right of the *For Each* component.
+. Set the *Value* field to `Multicast Ok`.
+
+== Run and Test your Mule Application
```

Comment:
```suggestion
== Run and Test Your Mule Application
```

Reply:
Are we running it with the purpose of testing it?
Otherwise, it is more logical to test first and then run...or deploy.